### DISCUSSION TITLE
NextJS - Hot reload event? how can i listen for this to close active connections

### DISCUSSION THREAD

**Discussion by ShanonJackson** [vercel/next.js#15654] [Help]
```
const Stream = // doesn't matter
(async () => {
    await LiveStream.connect();
    LiveStream.subscribe((value) => Stream.concat(value))
})()

export default (req, res) => res.write(Stream.toString()) // ignore the code here it doesn't matter just using as example.
```
Where can i put a `LiveStream.disconnect()` event? Everytime this file refreshes NextJS (dev) opens a "NEW" connection but there seems to be no event fired (I.E none of these `[exit, SIGINT, SIGUSR1, SIGUSR2, uncaughtException, SIGTERM]`);

NOTE: My team has told me under no circumstance can i open a new connection for every request; i must open a single connection for ALL requests.

TL;DR How can i listen for NextJS Hot reload and close a live stream connection - please don't reply with suggestions of why this is the opposite of serverless.

**Comment by noclat** [comment]
Hello, same issue here, I'd need a hot reload event to kill some Redis clients in development otherwise they just pile up to the sky :/. Packages like Bull needs to create local clients for each queue.

**Reply by keepeek-rd** [reply]
I have the same issue with redis and a DB pool connection :(
